### ISSUE TITLE
BulkInsertOrUpdate does not use default generated values

### ISSUE THREAD

**Issue by bo100nka** [borisdj/EFCore.BulkExtensions#145]
In my EF.Core solution i got:

```
// Use SQL Server NEWID() FX for generating [Id]
entityTypeBuilder.Property(nameof(IEntity.Id))
	.HasDefaultValueSql("NEWID()")
	.ValueGeneratedOnAdd();

// Use SQL Server SYSDATETIMEOFFSET() FX for generating [CreatedOn]
entityTypeBuilder.Property(nameof(IEntity.CreatedOn))
	.HasColumnType("datetimeoffset(3)")
	.HasDefaultValueSql("SYSDATETIMEOFFSET()")
	.ValueGeneratedOnAdd();
```
So when I do:
```
dbContext.MyEntity.Add(new MyEntity(....));
dbContext.SaveChanges();
```
It will properly insert a new entry ot the database with ID and CreatedOn fields populated with NewId() and SysDateTimeOffset() respectively.

Which is expected behavior but using BulkInsertOrUpdate it doesn't seem to happen:
```
var id = Guid.NewGuid();

db.MyEntities.Add(new MyEntity("abc", "123", "xyz", new CultureInfo("en"), "val") { Id = id });
db.SaveChanges();

var overwriteEntity = new MyEntity("abc", "123", "xyz", new CultureInfo("en"), "val34") { Id = id };
var newEntity = new MyEntity("abc", "123", "xyz", new CultureInfo("de"), "de-val34");

using (var transaction = db.Database.BeginTransaction())
{
	db.BulkInsertOrUpdate(new List<MyEntity> { overwriteEntity, newEntity });
	transaction.Commit();
}
```
It does not fail, but the inserted values are:
```
Id	CreatedOn	EntityName	EntityId	AttributeName	Culture	Value
00000000-0000-0000-0000-000000000000	0001-01-01 00:00:00.000 +00:00	abc	123	xyz	de	de-val34
D19A2EE8-F5CE-4B44-A780-73593FB17B6C	0001-01-01 00:00:00.000 +00:00	abc	123	xyz	en	val34
```
The second row has the ID because i explicitly set it. The CreatedOn's field setter is with "internal" accessor so I cannot set it. Regardless of why i can't set it, the point is, it should use the default values specified by dbContext's model builder.

**Comment by borisdj** [comment]
You can achieve this by explicitly excluding these properties in BulkConfig:
```C#
var bulkConfig = new BulkConfig()
{
  PropertiesToExclude = new List<string> { nameof(MyClass.MyClassId), nameof(MyClass.CreatedOn) }
};
db.BulkInsertOrUpdate(new List<MyEntity> { overwriteEntity, newEntity }, bulkConfig);
```

**Comment by acordmike** [comment]
It would be possible to determine some of these fields based on the entity configuration/attributes.

**Comment by borisdj** [comment]
Will consider it.

**Comment by rdasgupta7141** [comment]
@borisdj : How can we pick and choose when we want db to generate and when the application supplying it. If I don't add the column in the excludeproperty and try to make it null following the screenshot below, it throws cannot be dbnull error
![image](https://user-images.githubusercontent.com/26131729/107063500-50e37400-678f-11eb-84b5-3ba2ee2a480e.png)

**Comment by borisdj** [comment]
I'm not sure if I understand exactly your use case.
You can neither set them, or configure them to be excluded.
Also now there are separate config options for Insert and for Update regarding Include/Exclude.

**Comment by borisdj** [comment]
Should be fixed with 3.4.4
